tools/freeze_requirements.py

```diff
@@ -0,0 +1,75 @@
+import os
+from concurrent.futures import ThreadPoolExecutor
+from subprocess import run
+from typing import Tuple
+
+from tools.lib import gitroot
+from tools.lib import ts_print as print
+
+
+def worker(cmd: Tuple[str, ...]) -> Tuple[str, int]:
+    cmd_pretty = " ".join(cmd)
+    print(f"+ {cmd_pretty}")
+    proc = run(cmd, capture_output=True)
+    rc = proc.returncode
+    print(f"+ {cmd_pretty} returned {rc}")
+    if rc != 0:
+        print(f"stdout: {proc.stdout.decode()}\nstderr: {proc.stderr.decode()}")
+    return cmd_pretty, rc
+
+
+def main() -> int:
+    os.chdir(gitroot())
```

Comment:
rather than chdir -- this can probably just join the root with the filenames passed to `pip-tools` -- chdir makes this more difficult to test and/or debug

Reply:
Oh yeah, I originally did this because I didn't want like, `/Users/josh/...` in the annotations but they've since been removed.